packages/kit/src/runtime/client/renderer.js

```diff
@@ -451,19 +444,19 @@ export class Renderer {
 			result.props[`props_${i}`] = loaded ? await loaded.props : null;
 		}
 
-		if (
-			!this.current.page ||
-			page.path !== this.current.page.path ||
-			page.query.toString() !== this.current.page.query.toString()
-		) {
-			result.props.page = page;
+		if (!this.current.url || url.href !== this.current.url.href) {
+			result.props.url = url;
+		}
+
+		if (!this.current.params || JSON.stringify(params) !== JSON.stringify(this.current.params)) {
+			result.props.params = params;
 		}
 
 		const leaf = filtered[filtered.length - 1];
 		const maxage = leaf.loaded && leaf.loaded.maxage;
 
 		if (maxage) {
-			const key = `${page.path}?${page.query}`;
+			const key = url.origin + url.pathname; // omit hash
```

Comment:
Do we want to include `+ url.search` here like we had before?

Reply:
whoops, that was meant to be `pathname + search` not `origin + pathname`, d'oh